### PR TITLE
Changes for grouped select and API integration

### PR DESCRIPTION
**Pull request by risabhbaheti**
-Made the grouped select component
- Integrated the API for getting infected count
-Added data for country and US state population

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import {stateData} from "./components/grouped-select/usa-state-population-data";
+import {countryData} from "./components/grouped-select/country-population-data";
```
Comment (ashubham) [inline src/api.ts]: Why do we import these from components ? These should live within this module

File: src/components/controls/controls.tsx [hunk 2]
```diff
@@ -100,7 +102,10 @@ function reducer(state, action) {
 
 export const Controls: React.FC<Props> = ({ onChange }) => {
     const classes = useStyles();
+    initialState.initialNumberOfCases = getCovidData(countryData[0].Country);
+    initialState.totalPopulation = getPopulationData(countryData[0].Country)
```
Comment (ashubham) [inline src/components/controls/controls.tsx]: No need to read from countryData. Hard code 'USA' here

File: src/components/controls/controls.tsx [hunk 3]
```diff
@@ -9,6 +9,10 @@ import {
   } from '@material-ui/pickers';
 import {Slider} from '../slider/slider';
 import {sliders} from './controls-data';
+import GroupedSelect from "../grouped-select/grouped-select";
+import {stateData} from "../grouped-select/usa-state-population-data";
```
Comment (ashubham) [inline src/components/controls/controls.tsx]: No used, not needed

File: src/components/grouped-select/grouped-select.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (ashubham) [inline src/components/grouped-select/grouped-select.tsx]: Rename this component to RegionSelect ?

File: src/components/grouped-select/grouped-select.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        },
+    }),
+);
+
+export default function GroupedSelect(props) {
+    const classes = useStyles();
+
+    return (
+        <div>
+            <FormControl className={classes.formControl}>
```
Comment (ashubham) [inline src/components/grouped-select/grouped-select.tsx]: Do we need FormControl wrapper ?

File: src/components/grouped-select/usa-state-population-data.ts [hunk 6]
```diff
@@ -0,0 +1,206 @@
+export const stateData = [
```
Comment (ashubham) [inline src/components/grouped-select/usa-state-population-data.ts]: use from `generated-data/country-state.json`

File: src/components/grouped-select/country-population-data.ts [hunk 7]
```diff
@@ -0,0 +1,222 @@
+export const countryData =[
```
Comment (ashubham) [inline src/components/grouped-select/country-population-data.ts]: use from `generated-data/country-state.json`

File: src/api.ts (hunk 1 above)
Comment (ashubham) [inline src/api.ts]: use from `generated-data/country-state.json`

File: src/components/controls/controls.tsx (hunk 2 above)
Comment (risabhbaheti) [reply src/components/controls/controls.tsx]: Done

File: src/components/controls/controls.tsx (hunk 3 above)
Comment (risabhbaheti) [reply src/components/controls/controls.tsx]: Done

File: src/components/grouped-select/grouped-select.tsx (hunk 4 above)
Comment (risabhbaheti) [reply src/components/grouped-select/grouped-select.tsx]: Done

File: src/components/grouped-select/usa-state-population-data.ts (hunk 6 above)
Comment (risabhbaheti) [reply src/components/grouped-select/usa-state-population-data.ts]: Removed this

File: src/components/grouped-select/country-population-data.ts (hunk 7 above)
Comment (risabhbaheti) [reply src/components/grouped-select/country-population-data.ts]: Removed this

**Pull request merged**